### DISCUSSION TITLE
Read processed data from xns11

### DISCUSSION THREAD

**Discussion by JulKoch** [DHI/mikeio1d#43] [Q&A] [answered]
Hi 
Is it possible to read the processed data table (level, cross section area, etc.) for the cross sections?
Help is much appreciated.

**Comment by JesperGr** [comment]
The mikeio1d will not help you do that. 

The MIKE 1D components are .NET components, so those data can be read by using the .NET API. This is also possible to use from within Python (using the pythonnet package). Though, there is no wrapper functionality in mikeio1d for that. 

For some examples on using the .NET API for cross sections, you may take a look at:
https://github.com/DHI/Mike1D-SDK-examples/blob/master/DHI.Mike1D.Examples/CrossSectionExamples.cs

**Reply by gedaskir** [reply, marked as answer]
Hi, JesperGr is right that mikeio1d will not be able to do that out of the box. However, the relevant .NET class instances are available in mikeio1d. For example, it is possible to read the processed data into pandas data frame using such script:

```python
from mikeio1d.xns11 import Xns11

def get_processed_data(xns, print_data_frame=False):
    processed_data = {}
    for xs in xns.file:
        df = pd.DataFrame()
        df['level'] = list(xs.BaseCrossSection.ProcessedLevels)
        df['area'] = list(xs.BaseCrossSection.ProcessedAreas)
        df['radius'] = list(xs.BaseCrossSection.ProcessedRadii)
        df['width'] = list(xs.BaseCrossSection.ProcessedStorageWidths)
        df['add. area'] = list(xs.BaseCrossSection.ProcessedAdditionalSurfaceAreas)
        df['resistance'] = list(xs.BaseCrossSection.ProcessedResistanceFactors)

        location = f'{xs.Location.ID}, {xs.Location.Chainage}'
        processed_data[location] = df

        if (print_data_frame):
            print(f'Cross-section location: {location}')
            print(df)
            print()
    
    return processed_data

xns = Xns11("FlowSplit.xns11")
processed_data = get_processed_data(xns, print_data_frame=True)

print('Access particular cross-section')
print(processed_data['Branch1, 0.0'])
```
